lib/Dialect/SPIRV/SPIRVOps.cpp

```diff
@@ -227,6 +227,33 @@ static LogicalResult verifyMemoryAccessAttribute(LoadStoreOpTy loadStoreOp) {
   return success();
 }
 
+template <typename BarrierOp>
+static LogicalResult verifyMemorySemantics(BarrierOp op) {
+  auto memorySemantics = op.memory_semantics();
+  SmallVector<spirv::MemorySemantics, 4> memorySemanticsBits{
+      spirv::MemorySemantics::Acquire, spirv::MemorySemantics::Release,
+      spirv::MemorySemantics::AcquireRelease,
+      spirv::MemorySemantics::SequentiallyConsistent};
+
+  uint32_t bitCount = 0;
+  for (auto &memorySemanticsBit : memorySemanticsBits) {
+    bitCount += static_cast<uint32_t>(
+        spirv::bitEnumContains(memorySemantics, memorySemanticsBit));
+  }
+
+  if (!bitCount) {
```

Comment:
Is this correct? What is the spec rule for this? I think we can have `None`/`Relaxed` memory semantics?

Reply:
@antiagainst oh, I missed that none is relaxed, sorry about it. Thanks for mention this!